nservicebus/vscode.md

```diff
@@ -0,0 +1,263 @@
+---
+title: Debugging NServiceBus in Visual Studio Code
+summary: How to configure Visual Studio Code to build and debug multiple NServiceBus endoints simultaneously
+component: Core
+reviewed: 2018-08-01
+---
+
+This article describes how to configure VSCode to build an NServiceBus solution with multiple projects and debug multiple endpoints simulateously.
+
+NServiceBus 7 and above is designed to run on .NET Core, which means instead of using Visual Studio as an IDE, users can use the cross-platform [Visual Studio Code](https://code.visualstudio.com/) application (or "VSCode") to build systems with NServiceBus.
+
+While Visual Studio is designed only to run .NET projects, VSCode is a more general-purpose IDE for any language, so some configuration is necessary for VSCode to know how to build and debug .NET projects. Because NServiceBus projects typically involve running multiple startup projects to test multiple endpoints simultaneously, a little extra configuration is needed.
+
+## Prerequisities
+
+* This article assumes knowledge of NServiceBus solutions.
+* VSCode must have the [C# for Visual Studio (powered by OmniSharp)](https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp) extension installed.
+
+## VSCode configuration files
+
+The build and debug system in VSCode is controlled by two files located in a `.vscode` directory at the project root:
+
+* _{Project Root}_
+    * `.vscode/`
+        * `launch.json`
+        * `tasks.json`
+
+The **launch.json** file describes different projects within the solution, and describes how to launch each one.
+
+The **tasks.json** file describes actions to take, the most common of which is to build the code, which can be executed as a prerequisite of executing a configuration in **launch.json**.
+
+## tasks.json
+
+The simplest **tasks.json** file will describe how to build the solution using `dotnet build`:
+
+```json
+{
+    "version": "2.0.0",
+    "tasks": [
+        {
+            "label": "build",
+            "command": "dotnet",
+            "type": "process",
+            "args": [
+                "build",
+                "${workspaceFolder}/path/to/SolutionName.sln"
+            ],
+            "problemMatcher": "$msCompile"
+        }
+    ]
+}
+```
+
+For an NServiceBus solution, use the content above and adjust the path of the `sln` file.
+
+For more information on **tasks.json** see [Visual Studio Code: Integrate with External Tools via Tasks](https://code.visualstudio.com/docs/editor/tasks).
+
+## launch.json
+
+The high-level structure of the launch.json file contains a collection of individual project objects in `configurations`, and then (optionally) a `compounds` collection that group up multiple configurations that will be launched together.
```

Comment:
Proposed change:

The high-level structure of the **launch.json** file contains a collection of individual project objects in `configurations` and an optional `compounds` collection that lists multiple configurations that should be launched at the same time.